utest/TestSelfUpdateScenarios.cpp

```diff
@@ -1,336 +1,406 @@
+                {
+                    "activityId": "random-uuid-as-string",
+                    "timestamp": 123456789
+                }
+            )";
+            // clang-format on
+
+            execute(sua::IMqttProcessor::TOPIC_STATE_GET, json);
         }
 
-        sua::Context & ctx() {
-            return sua.context();
+        void execute(const std::string& topic, const std::string& json) {
+            std::stringstream ss;
+            ss << "mosquitto_pub";
+            ss << " -h " << testBrokerHost;
+            ss << " -t " << topic;
+            ss << " -m '" << json << "'";
+            std::system(ss.str().c_str());
+            std::this_thread::sleep_for(1s);
         }
 
         void SetUp() override {
-            sua::Logger::instance().setLogLevel(sua::Logger::Level::None);
+            sua::Logger::instance().init();
+            sua::Logger::instance().setLogLevel(sua::Logger::Level::All);
 
             downloader = std::make_shared<MockDownloader>();
             ctx().downloaderAgent = downloader;
 
-            fsm = std::make_shared<MockFSM>(sua.context());
+            fsm = std::make_shared<MockFSM>(sua.context(), visitedStates);
             ctx().stateMachine = fsm;
 
-            messagingProtocol = std::make_shared<MockMqttMessagingProtocol>();
-            ctx().messagingProtocol = messagingProtocol;
-
-            mqttProcessor = std::make_shared<MockMqttProcessor>();
-            ctx().mqttProcessor = mqttProcessor;
-
             installerAgent = std::make_shared<MockRaucInstaller>();
             ctx().installerAgent = installerAgent;
-        
-            ctx().bundleChecker = std::make_shared<sua::BundleChecker>();
+ 
+            mqttProcessor = std::make_shared<MockMqttProcessor>(sua.context(), sentMessages);
+            ctx().mqttProcessor = mqttProcessor;
+
+            ctx().messagingProtocol = std::make_shared<sua::MqttMessagingProtocolJSON>();
+            ctx().bundleChecker     = std::make_shared<sua::BundleChecker>();
         }
 
         sua::SelfUpdateAgent sua;
 
-        std::shared_ptr<MockFSM>                   fsm;
-        std::shared_ptr<MockMqttMessagingProtocol> messagingProtocol;
-        std::shared_ptr<MockMqttProcessor>         mqttProcessor;
-        std::shared_ptr<MockRaucInstaller>         installerAgent;
-        std::shared_ptr<MockDownloader>            downloader;
+        std::shared_ptr<MockFSM>           fsm;
+        std::shared_ptr<MockMqttProcessor> mqttProcessor;
+        std::shared_ptr<MockRaucInstaller> installerAgent;
+        std::shared_ptr<MockDownloader>    downloader;
+
+        std::string testBrokerHost = "localhost";
+        int         testBrokerPort = 1883;
+
+        std::string bundleUnderTest;
+
+        std::vector<std::string> visitedStates;
+        std::vector<std::string> expectedStates;
+
+        std::vector<sua::MqttMessage> sentMessages;
+        std::vector<sua::MqttMessage> expectedMessages;
```

Comment:
collecting the states and messages as sequence is a good concept for easy check :-)